### DISCUSSION TITLE
Proposed change for new api-client syntax regarding query params

### DISCUSSION THREAD

**Discussion by niklaswolf** [shopware/frontends#574] [Ideas]
While moving to the new API-client we stumbled upon the syntax how to "declare" query params.
Currently you have to do it like this:
```
/my/path?name,id

{
    name: "my-name",
    id: "my-id"
}
```

We'd like to propose changing it to the following syntax:
```
/my/path?name={name}&id={id}

{
    name: "my-name",
    id: "my-id"
}
```

**Advantages:**

- consistent usage of placeholders, because if you have a parameter inside the path directly, you also specify them using `{}`, e.g. `/document/download/{documentId}/{deepLinkCode}`
- generally more understandable because this is how query-parameters are written in URLs. So not so much Domain-specific-language, but more aligned to the URL-Standard
- more flexibility, because you are free to name the placeholder-variable something other than the query-param key.

**Disadvantages:**

- more verbose

**Comment by patzick** [comment]
as with #871 - query params are now removed from the path definitions. The only variables that stays are pathParams. All other params, endpoint headers or body are added in type definitions. Example definition of the kind of endpoint:

```ts
export type operations = {
  // ....
"readCategory post /category/{navigationId}": {
    contentType?: "application/json";
    accept?: "application/json";
    headers?: {
      /** Instructs Shopware to try and resolve SEO URLs for the given navigation item */
      "sw-include-seo-urls"?: boolean; // TODO: [OpenAPI][readCategory] - add header to the parameters
    };
    query?: {
      /** Resolves only the given slot identifiers. The identifiers have to be seperated by a '|' character */
      slots?: string;
    };
    pathParams: {
      /** Identifier of the category to be fetched */
      navigationId: string;
    };
    body: components["schemas"]["Criteria"] &
      components["schemas"]["ProductListingCriteria"];
    response: components["schemas"]["Category"];
    responseCode: 200;
  };
//....
}
```


and the invocation looks like 

```ts
const result = await apiClient.invoke(
      "readCategory post /category/{navigationId}",
      {
        pathParams: {
          navigationId: categoryId,
        },
        headers: {
          "sw-include-seo-urls": true,
        },
        body: {
          // some criteria
        },
      },
    );
```

**Reply by niklaswolf** [reply]
Even better :)
